### PR TITLE
add remove_bins_from_binned_contig method

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

**Comment by Tianhao-Gu** [comment]
@msneddon 
adding method remove_bins_from_binned_contig: remove a list of bins from BinnedContig object

File: lib/MetagenomeUtils/Utils/MetagenomeFileUtils.py [hunk 1]
```diff
@@ -339,6 +388,109 @@ def _generate_report(self, generated_assembly_ref_list, params):
...
+            total_gc_count += round(sum_contig_len * bin.get('gc'), 5)
+            total_cov_len += round(sum_contig_len * bin.get('cov'), 5)
+
+        contig_bin = {
+            'bid': new_bin_id,
+            'contigs': total_contigs,
+            'n_contigs': len(total_contigs),
+            'gc': round(float(total_gc_count)/total_sum_contig_len, 5),
+            'sum_contig_len': total_sum_contig_len,
+            'cov': round(float(total_cov_len)/total_sum_contig_len, 5)
```
Comment (Tianhao-Gu) [inline lib/MetagenomeUtils/Utils/MetagenomeFileUtils.py]: could you please verify this is right logic for contig coverage when merge?

Comment (msneddon) [inline lib/MetagenomeUtils/Utils/MetagenomeFileUtils.py]: The math (computing the weighted average for cov and GC) looks correct to me!  One minor point is that we probably shouldn't round the calculations on 434 and 435- just compute to full precision there and round at the end when we're saving the final value.

Comment (Tianhao-Gu) [inline lib/MetagenomeUtils/Utils/MetagenomeFileUtils.py]: @msneddon 👍 Cool. Removed rounding during computing.

**Comment by msneddon** [comment]
Great!

**Pull request merged**